### PR TITLE
Feature/maintenance

### PR DESCRIPTION
**Pull request by yiliansource**

### CODE REVIEW

File: Controllers/AdministrationController.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            return View();
+        }
+
+        [HttpPost]
+        [Route("administration/maintenance/{operation}")]
+        public IActionResult ProcessMaintenanceOperation(string operation)
+        {
+            _configuration[MaintenanceModeConstants.ConfigurationKey] = operation.Equals("activate")
+                ? true.ToString()
+                : operation.Equals("deactivate")
```
Comment (AquaXV) [inline Controllers/AdministrationController.cs]: nested ternary operators are hard to read and give a high maintenance penalty (heh, get it, maintenance...)

File: Core/Maintenance/MaintenanceModeMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                // If the maintenance is active and the user does not have the access rights to the rest of the page, redirect him to the maintenance page.
+                if (isMaintenance && !isUserSiteAdmin)
+                {
+                    context.Response.Redirect(MaintenanceModeConstants.PagePath);
+                }
+            }
+            else
+            {
+                // If the user is already on the maintenance page, but the maintenance is not enabled, redirect him to the homepage.
+                if (!isMaintenance)
```
Comment (AquaXV) [inline Core/Maintenance/MaintenanceModeMiddleware.cs:43]: You could optionally merge else and if into else if

File: Controllers/AdministrationController.cs (hunk 1 above)
Comment (yiliansource) [reply Controllers/AdministrationController.cs]: Got it, I'll think up something cleaner.

File: Core/Maintenance/MaintenanceModeMiddleware.cs (hunk 2 above)
Comment (yiliansource) [reply Core/Maintenance/MaintenanceModeMiddleware.cs:43]: Eh, I think I'll keep it like this, feels a bit more readable.

**Review by AquaXV — APPROVED** [review]

**Pull request merged**